core/services/job/orm.go

```diff
@@ -240,42 +242,64 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 				}
 			}
 
-			if jb.OCR2OracleSpec.PluginType == Mercury {
-				if jb.OCR2OracleSpec.FeedID == (common.Hash{}) {
-					return errors.New("feed ID is required for mercury plugin type")
-				}
-			} else {
-				if jb.OCR2OracleSpec.FeedID != (common.Hash{}) {
-					return errors.New("feed ID is not currently supported for non-mercury jobs")
+			sendingKeysDefined, err := areSendingKeysDefined(jb.OCR2OracleSpec)
+			if err != nil {
+				return err
+			}
+
+			if !sendingKeysDefined && !jb.OCR2OracleSpec.TransmitterID.Valid {
+				return errors.New("neither sending keys nor transmitter ID is defined")
+			}
+
+			if sendingKeysDefined && jb.OCR2OracleSpec.TransmitterID.Valid {
+				return errors.New("sending keys and transmitter ID can't both be defined")
```

Comment:
Maybe extract this to another function for validatind sendingKeys and transmitterID, Idk what I would call it though

Reply:
I think it looks good here but I'll leave it up to you